OpenRA.Mods.Common/Traits/Player/TimeLimitManager.cs

```diff
@@ -49,6 +49,16 @@ public class TimeLimitManagerInfo : ITraitInfo, ILobbyOptions
 		[Desc("Display order for the time limit dropdown in the lobby.")]
 		public readonly int TimeLimitDisplayOrder = 0;
 
+		[Desc("Will prevent showing/playing the built-in time limit warnings when set to true.")]
+		public readonly bool SkipTimeRemainingNotifications = false;
+
+		[Desc("Will prevent showing/playing the built-in timer expired notification when set to true.")]
+		public readonly bool SkipTimerExpiredNotification = false;
+
+		[Desc("If set to true, will make all players lose by default when the timer expires.",
+			"Disable this if you want to handle the WinStates externally, for example from a Lua script.")]
+		public readonly bool UpdatePlayerWinStates = true;
```

Comment:
IMO it would be much nicer if we could detect this automatically based on the current victory conditions.
What do you think about adding a `GameTimerExpired` method to `INotifyObjectivesUpdated` instead?

This would then help with my suggestion above for setting the winner based on the highest score for `ConquestVictoryConditions`, `StrategicVictoryConditions` could do it based on whoever has the most control points, and scripted missions fire the lua trigger.